gufe/protocols/protocol_settings.py

```diff
@@ -0,0 +1,381 @@
+from typing import Optional, Dict, Union, List
+
+from pydantic import BaseModel, validator
+from openff.units import unit
+
+
+class SystemSettings(BaseModel):
+    """Settings describing the simulation system settings.
+
+    Attributes
+    ----------
+    nonbonded_method : str
+        Which nonbonded electrostatic method to use, currently only PME
+        is supported.
+    nonbonded_cutoff : float * unit.nanometer
+        Cutoff value for short range interactions.
+        Default 1.0 * unit.nanometer.
+    constraints : str
+        Which bonds and angles should be constrained. Default None.
+    rigid_water : bool
+        Whether to apply rigid constraints to water molecules. Default True.
+    hydrogen_mass : float
+        How much mass to repartition to hydrogen. Default None, no
+        repartitioning will occur.
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    nonbonded_method = 'PME'
+    nonbonded_cutoff = 1.0 * unit.nanometer
+    constraints: Union[str, None] = 'HBonds'  # Usually use HBonds
+    rigid_water = True
+    remove_com = True  # Probably want False here
+    hydrogen_mass: Union[float, None] = None
+
+
+class TopologySettings(BaseModel):
+    """Settings for creating Topologies for each component
```

Comment:
Can this be more clearly defined? I'm not clear on what this class is supposed to represent from the brief doctstring.